### PR TITLE
Portal + front-end Auth0 integration and basic Vue routing

### PR DESCRIPTION
**Pull request by AndrewGibson27**
#### What's this PR do?
+ Lots of file reorganization
+ Integrates log in/log out flow via Auth0
+ Adds Auth0 access token to all API requests
+ Adds basic Vue routing for future SPA

NOTE: It'll look hideous! No styles have been applied.

#### Why are we doing this? How does it help us?
Just UMPing along.

#### How should this be manually tested?
*You'll need a non-staff Auth0 account in our test tenant for this (and future UMP reviews)*

+ Ensure you have test values for `AUTH0_CLIENT_ID`, `AUTH0_CLIENT_SECRET` and `AUTH0_DOMAIN` in `env-docker`
+ Ensure you're logged out of any non-staff accounts in our Auth0 test tenant
+ `make`
+ `make restart` (wait for Webpack to finish)
+ Go to `/user-portal`. Confirm it redirects you to our universal log-in page.
+ Log in. Confirm you get redirected back to `/user-portal`, where the page will hang for a few (eventually it'll show a loading icon) while it checks your Auth0 session. After a few, you should see "this is home" and a few links.
+ Click those links. Confirm the text changes along with the URL, indicating the client-side routing is working.
+ Open up Chrome developer tools. Go to the network tab, then click "XHR" below it. Do a hard refresh, then look for a GET request to `content` (a temporary API call I dropped in for PR-review purposes). Click into it and look for "request headers." One should be `Authorization: Bearer <random string>`. That means our Auth0 access token is getting sent along with API calls.
+ Click "log out." You should get redirected to `/donate` afterward (that flow is up for debate). Go back to `/user-portal` and confirm you once again get prompted to log in.

#### How should this change be communicated to end users?
NA.

#### Are there any smells or added technical debt to note?
NA. Question for @x110dc: Right now, every API call will include the Auth0 access token in the Authorization header. Is there any security danger in that if, for example, we ever need to call a non-Texas Tribune API in this app? Can third parties do anything with that access token?

#### What are the relevant tickets?
https://3.basecamp.com/3098728/buckets/10357222/todos/1720701890

#### Have you done the following, if applicable:
* [ ] Added automated tests? *(later)*
* [ ] Tested manually on mobile? *(will do later when app is further along)*
* [ ] Checked for performance implications?
* [ ] Checked for security implications? *(see question above)*
* [ ] Updated the documentation/wiki? *(later)*

#### TODOs / next steps:
Formalize Vue route-based data fetching.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,13 +22,15 @@
...
     "@babel/runtime": "^7.4.3",
+    "auth0-js": "^9.10.2",
     "autoprefixer": "^9.4.2",
     "axios": "^0.18.0",
     "babel-loader": "^8.0.0",
     "css-loader": "^0.28.11",
     "es6-promise": "^4.2.4",
     "fast-async": "7",
     "file-loader": "^1.1.11",
+    "fontfaceobserver": "^2.1.0",
```
Comment (ashley-hebler) [inline package.json]: I like the idea of having this included this way, but why do we have it in `font_load.html` and here? Maybe I'm misunderstanding how that works.

File: static/js/src/entry/circle/TopForm.vue [hunk 2]
```diff
@@ -161,15 +161,15 @@
...
 <script>
-import Hidden from '../../elements/Hidden.vue';
-import LocalHidden from '../../elements/LocalHidden.vue';
-import PayFees from '../../elements/PayFees.vue';
-import TextInput from '../../elements/TextInput.vue';
-import ManualPay from '../../elements/ManualPay.vue';
-import ManualSubmit from '../../elements/ManualSubmit.vue';
-import NativePay from '../../elements/NativePay.vue';
+import Hidden from '../../connected-elements/Hidden.vue';
+import LocalHidden from '../../local-elements/Hidden.vue';
```
Comment (ashley-hebler) [inline static/js/src/entry/circle/TopForm.vue]: here, are we saying connected means hooked up to some API? And local is just for our app?

File: static/js/src/entry/user-portal/constants.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export const LOGIN_REDIRECT_KEY = 'postLoginRedirect';
+export const LOGGED_IN_FLAG_KEY = 'isLoggedIn';
+export const NON_STAFF_CONNECTION = 'Username-Password-Authentication';
+export const AUTH_DOMAIN = process.env.AUTH0_DOMAIN;
```
Comment (ashley-hebler) [inline static/js/src/entry/user-portal/constants.js]: Might be helpful to [document](https://github.com/texastribune/donations#environment) these for easy troubleshooting down the line. Mine barked at me in my console so it was pretty clear, but might help for someone setting up a new env.

**Comment by ashley-hebler** [comment]
Woo! Enjoying this SPA lyfe.
All worked great for me. That's awesome that you already have routing plugged in. I left a few questions/comments but nothing that blocks this.

This is such an edge case, but I wanted to note it. I _signed up_  initially as opposed to _logged in_ (because of my bad login org) and it wouldn't actually let me see `/user-portal` until I verified. It redirected to `/donate`, which makes sense because I wasn't in a real logged in state

File: package.json (hunk 1 above)
Comment (AndrewGibson27) [reply package.json]: It's really just for documentation's sake. To be aware of what version we are using. There's no direct connection between what's inlined in `font_load.html` and what lives in `node_modules` after a `yarn`.

File: static/js/src/entry/circle/TopForm.vue (hunk 2 above)
Comment (AndrewGibson27) [reply static/js/src/entry/circle/TopForm.vue]: Connected means connected to the Vuex store. Local means relying on local component state. I'll document that.

File: static/js/src/entry/user-portal/constants.js (hunk 3 above)
Comment (AndrewGibson27) [reply static/js/src/entry/user-portal/constants.js]: I shall!

File: static/js/src/entry/circle/TopForm.vue (hunk 2 above)
Comment (ashley-hebler) [reply static/js/src/entry/circle/TopForm.vue]: Noted!

**Comment by AndrewGibson27** [comment]
@x110dc Looks like Webpack hadn't finished its build. If you use the Heroku URL, you won't have to worry about that. (Or if you want to try locally, do `make restart` and wait for Webpack to spit out lots of green and blue text in your terminal.)

**Comment by x110dc** [comment]
Hmm. My fault -- wasn't running `yarn run dev` but now `yarn` appears to be doing it's thing but I never get blue and green text (maybe it's just my terminal) but when I hurt the URL I just get a blank page -- no redirect. I'll try Heroku -- but I'll need to have this working locally at some point.

**Comment by AndrewGibson27** [comment]
Let's look at this together. My instructions were probably bad.

**Comment by x110dc** [comment]
Worked fine locally. LGTM.

**Pull request merged**